### PR TITLE
[sharktank] Add perplexity CI to sharktank dashboard

### PR DESCRIPTION
**Pull request by archana-ramalingam**
Add perplexity CI to sharktank dashboard
Rename _vmfb to _iree
Add Perplexity scoreboard & description
Add descriptive errors for better logging

### CODE REVIEW

**Comment by archana-ramalingam** [comment]
@saienduri Would like to add Perplexity CI results to this [dashboard](https://nod-ai.github.io/SHARK-Platform/?sort=result) too. pytest-html doesn't auto-merge with existing html files. Do we need a utility function to do this separately?

File: .github/workflows/ci_eval.yaml [hunk 1]
```diff
@@ -7,6 +7,7 @@
 name: CI - Perplexity
 
 on:
+  pull_request:
```
Comment (marbre) [inline .github/workflows/ci_eval.yaml:10]: I assume we don't want to run this on every PR as it takes quite some time?

File: .github/workflows/ci_eval.yaml [hunk 2]
```diff
@@ -122,5 +129,11 @@ jobs:
           pip install --no-compile -f https://iree.dev/pip-release-links.html --src deps \
             -e "git+https://github.com/iree-org/iree-turbine.git#egg=iree-turbine"
 
-      - name: Run perplexity test in eager mode
-        run:  pytest -n 8 -v -s sharktank/tests/evaluate/perplexity_torch_test.py --longrun --llama3-8b-f16-model-path=/data/llama3.1/8b/llama8b_f16.irpa --llama3-8b-tokenizer-path=/data/llama3.1/8b/tokenizer_config.json
+      - name: Run perplexity test with Torch
+        run:  pytest -n 8 -v -s sharktank/tests/evaluate/perplexity_torch_test.py --longrun --llama3-8b-f16-model-path=/data/llama3.1/8b/llama8b_f16.irpa --llama3-8b-tokenizer-path=/data/llama3.1/8b/tokenizer_config.json --html=perplexity/perplexity_torch.html
+
+      - name: Deploy to GitHub Pages
+        uses: peaceiris/actions-gh-pages@v3
```
Comment (marbre) [inline .github/workflows/ci_eval.yaml:136]:
```suggestion
        uses: peaceiris/actions-gh-pages@4f9cc6602d3f66b9c108549d475ec49e8ef4d45e # v4.0.0
```
Actions should be pinned as as suggested by OpenSSF Scorecard, see https://github.com/ossf/scorecard/blob/main/docs/checks.md#pinned-dependencies.

Comment (marbre) [reply .github/workflows/ci_eval.yaml:136]: Furthermore, is it intended to push to gh-pages with every run of the CI? This probably creates the same problems outlined in issue #395.

File: .github/workflows/ci_eval.yaml [hunk 3]
```diff
@@ -74,12 +75,18 @@ jobs:
             iree-base-runtime \
             "numpy<2.0"
 
-      - name: Run perplexity test with vmfb
-        run:  pytest -n 8 -v -s sharktank/tests/evaluate/perplexity_vmfb_test.py --longrun --iree-device='hip://7' --iree-hip-target=gfx942 --iree-hal-target-backends=rocm --llama3-8b-f16-model-path=/data/llama3.1/8b/llama8b_f16.irpa --llama3-8b-tokenizer-path=/data/llama3.1/8b/tokenizer_config.json
+      - name: Run perplexity test with IREE
+        run:  pytest -n 8 -v -s sharktank/tests/evaluate/perplexity_iree_test.py --longrun --iree-device='hip://7' --iree-hip-target=gfx942 --iree-hal-target-backends=rocm --llama3-8b-f16-model-path=/data/llama3.1/8b/llama8b_f16.irpa --llama3-8b-tokenizer-path=/data/llama3.1/8b/tokenizer_config.json --html=perplexity/perplexity_iree.html
+
+      - name: Deploy to GitHub Pages
+        uses: peaceiris/actions-gh-pages@v3
```
Comment (marbre) [inline .github/workflows/ci_eval.yaml:82]:
```suggestion
        uses: peaceiris/actions-gh-pages@4f9cc6602d3f66b9c108549d475ec49e8ef4d45e # v4.0.0
```

File: .github/workflows/ci_eval.yaml (hunk 1 above)
Comment (archana-ramalingam) [reply .github/workflows/ci_eval.yaml:10]: Yes, this is only added to debug.

**Comment by saienduri** [comment]
I don't think we need to merge the html files themselves. Each one will live on it's own subsite with an overarching index.html that points to each of them was what I was thinking. I am planning to explore this when I get some time today

File: .github/workflows/ci_eval.yaml (hunk 2 above)
Comment (saienduri) [reply .github/workflows/ci_eval.yaml]: This is a nightly, so I don't think it will pose any presubmit problems

**Comment by archana-ramalingam** [comment]
> I don't think we need to merge the html files themselves. Each one will live on it's own subsite with an overarching index.html that points to each of them was what I was thinking. I am planning to explore this when I get some time today

Makes sense. I tried to push the `perplexity.html` to a new folder `perplexity`. But it seems to have replaced the existing [index.html](https://github.com/nod-ai/SHARK-Platform/commit/af71e4077a5e0f64371b34ce31f40aba8f2e0907) at root, the last time I ran it. 

Each folder in `gh-pages` branch rep a subsite IIUC. We need IREE perplexity and Torch perplexity as the subsites/folders.

File: sharktank/sharktank/evaluate/README.md [hunk 4]
```diff
@@ -9,16 +9,32 @@ pip install -r sharktank/requirements-tests.txt
...
+
+| CPU            | GPU        |
+|:-------------: |:----------:|
+| AMD EPYC 9554  | MI300X     |
+
+#### LLaMA 3.1
+
+|Models   |Model size (GB) |Torch      |IREE       |
+|:--------|:---------------|:----------|:----------|
+|8B f16   |16.07           |14.930181  |14.991893  |
```
Comment (sogartar) [inline sharktank/sharktank/evaluate/README.md:40]: Shouldn't these values be taken programmatically from the last perplexity run?

**Review by sogartar — APPROVED** [review]

File: sharktank/sharktank/evaluate/README.md (hunk 4 above)
Comment (archana-ramalingam) [reply sharktank/sharktank/evaluate/README.md:40]: I agree, but ideally these numbers are the best baseline numbers we compare against. We would want to update them only if we get a better score, not if it regressed. Hence, it's not programmatic.

**Pull request merged**